tools/pythonpkg/src/python_import_cache.cpp

```diff
@@ -10,12 +10,28 @@ py::handle PythonImportCacheItem::operator()(void) const {
 	return object;
 }
 
+bool PythonImportCacheItem::IsInstance(py::handle object) const {
+	auto type = (*this)();
+	if (type.ptr() == nullptr) {
+		//! Type was not imported
+		return false;
+	}
+	return py::isinstance(object, type);
+}
+
 PyObject *PythonImportCacheItem::AddCache(PythonImportCache &cache, py::object object) {
 	return cache.AddCache(move(object));
 }
 
 void PythonImportCacheItem::LoadModule(const string &name, PythonImportCache &cache) {
-	object = AddCache(cache, move(py::module::import(name.c_str())));
+	try {
+		object = AddCache(cache, move(py::module::import(name.c_str())));
+	} catch (py::error_already_set &e) {
+		if (IsRequired()) {
+			throw InvalidInputException("Required module '%s' failed to import", name);
+		}
+		return;
+	}
```

Comment:
I like this, I think this is really elegant :-)

Reply:
I think I'm changing it to a static method, and replacing all `py::isinstance` with it, maybe make a `IsInstanceUnsafe` to avoid the nullptr check where possible